crates/ere-nexus/src/lib.rs

```diff
@@ -0,0 +1,197 @@
+use std::path::PathBuf;
+use std::time::Instant;
+
+use nexus_sdk::compile::cargo::CargoPackager;
+use nexus_sdk::compile::{Compile, Compiler as NexusCompiler};
+use nexus_sdk::stwo::seq::{Proof, Stwo};
+use nexus_sdk::{Local, Prover, Verifiable};
+use tracing::info;
+use zkvm_interface::{
+    Compiler, Input, ProgramExecutionReport, ProgramProvingReport, ProverResourceType, zkVM,
+    zkVMError,
+};
+
+mod error;
+use crate::error::ProveError;
+use error::{CompileError, NexusError, VerifyError};
+
+/// NOTE: nexus's guest package name must be "ere-nexus-guest"
+/// TODO: This should be configurable in the future, but for now we hardcode it.
+const PACKAGE: &str = "ere-nexus-guest";
+
+#[allow(non_camel_case_types)]
+pub struct NEXUS_TARGET;
+
+impl Compiler for NEXUS_TARGET {
+    type Error = NexusError;
+
+    type Program = PathBuf;
+
+    fn compile(_: &std::path::Path) -> Result<Self::Program, Self::Error> {
+        let mut prover_compiler = NexusCompiler::<CargoPackager>::new(PACKAGE);
+        let elf_path = prover_compiler
+            .build()
+            .map_err(|e| CompileError::Client(e.into()))?;
+
+        Ok(elf_path)
+    }
+}
+
+pub struct EreNexus {
+    program: <NEXUS_TARGET as Compiler>::Program,
+}
+
+impl EreNexus {
+    pub fn new(
+        program: <NEXUS_TARGET as Compiler>::Program,
+        _resource_type: ProverResourceType,
+    ) -> Self {
+        Self { program }
+    }
+}
+impl zkVM for EreNexus {
+    fn execute(&self, inputs: &Input) -> Result<zkvm_interface::ProgramExecutionReport, zkVMError> {
+        let start = Instant::now();
+        // let private_input =  vec![] ;
+        let mut public_input = vec![];
+        for input in inputs.iter() {
+            public_input.extend(
+                input
+                    .as_bytes()
+                    .map_err(|e| NexusError::Prove(ProveError::Client(e.into())))
+                    .map_err(zkVMError::from)?,
+            );
+        }
+
+        Ok(ProgramExecutionReport {
+            execution_duration: start.elapsed(),
+            ..Default::default()
+        })
+    }
+
+    fn prove(
+        &self,
+        inputs: &Input,
+    ) -> Result<(Vec<u8>, zkvm_interface::ProgramProvingReport), zkVMError> {
+        let prover: Stwo<Local> = Stwo::new_from_file(&self.program.to_string_lossy().to_string())
+            .map_err(|e| NexusError::Prove(ProveError::Client(e.into())))
+            .map_err(zkVMError::from)?;
+
+        // let private_input =  vec![] ;
```

Comment:
can you leave a comment for this?

One convention that may be useful for simplifying the design is that all inputs to the vm are private and all outputs are public. If an input should be public, then it could just be returned from the function.